moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -69,25 +67,29 @@ bool RuckigSmoothing::applySmoothing(robot_trajectory::RobotTrajectory& trajecto
   const size_t num_waypoints = trajectory.getWayPointCount();
   if (num_waypoints < 2)
   {
-    RCLCPP_ERROR(LOGGER, "Trajectory does not have enough points to smooth with Ruckig");
-    return false;
+    RCLCPP_WARN(LOGGER,
+                "Trajectory does not have enough points to smooth with Ruckig. Returning an unmodified trajectory.");
+    return true;
   }
 
+  // Cache the trajectory in case we need to reset it
+  robot_trajectory::RobotTrajectory original_trajectory = trajectory;
+
   const size_t num_dof = group->getVariableCount();
 
   // This lib does not actually work properly when angles wrap around, so we need to unwind the path first
   trajectory.unwind();
 
   // Instantiate the smoother
-  double timestep = trajectory.getAverageSegmentDuration();
-  std::unique_ptr<ruckig::Ruckig<0>> ruckig_ptr;
-  ruckig_ptr = std::make_unique<ruckig::Ruckig<0>>(num_dof, timestep);
-  ruckig::InputParameter<0> ruckig_input{ num_dof };
-  ruckig::OutputParameter<0> ruckig_output{ num_dof };
+  double timestep = trajectory.getWayPointDurationFromStart(num_waypoints - 1) / (trajectory.getWayPointCount() - 1);
```

Comment:
@stephanie-eng  actually this change caused CI to fail. I'm not sure why but I trust one of the CI tests a lot now, since it compares against an analytical solution. (The duration was off by a factor of 2.) I'll look into it more tomorrow.

Reply:
Resolved -- the first waypoint in the test had a nonzero duration. So it was not an issue with `getAverageSegmentDuration()`

trajectory_->addSuffixWayPoint(robot_state, DEFAULT_TIMESTEP);